music21/metadata/__init__.py

```diff
@@ -355,22 +356,24 @@ def getAllContributorNamedValues(self) -> t.List[t.Tuple[str, t.Any]]:
         The tuple's first element will be of the form 'namespace:name'.
 
         >>> md = metadata.Metadata()
-        >>> md.addCustom('composer', 'Not a contributor')
-        >>> md.add('title', '[title of show]')
-        >>> md.addCustom('excerpt-start-measure', 1234)
-        >>> md.add('composer', ['Ludwig von Beethoven', 'Wolfgang Amadeus Mozart'])
+        >>> md['title'] = ['Dimitrij', 'False Dmitry']
+        >>> md.composer = 'Antonín Dvořák'
+        >>> md.opusNumber = 64
         >>> md.add('librettist', 'Marie Červinková-Riegrová')
+        >>> md['otherContributor'] = (metadata.Contributor(role='based on plot by',
+        ...                               name ='Ferdinand Mikovec'),
+        ...                           metadata.Contributor(role='original incomplete plot by',
+        ...                               name='Friedrich Schiller'))
+        >>> md.addCustom('composer', 'Not a contributor')
         >>> all = md.getAllContributorNamedValues()
-        >>> len(all)
-        3
-        >>> all[0]
-        ('marcrel:CMP', <music21.metadata.primitives.Contributor composer:Ludwig von Beethoven>)
-        >>> all[1]
-        ('marcrel:CMP', <music21.metadata.primitives.Contributor composer:Wolfgang Amadeus Mozart>)
-        >>> all[2][0]
-        'marcrel:LBT'
-        >>> all[2][1]
-        <music21.metadata.primitives.Contributor librettist:Marie Červinková-Riegrová>
+        >>> all
+        [('marcrel:CMP', <music21.metadata.primitives.Contributor composer:Antonín Dvořák>),
+        ('marcrel:LBT',
+        <music21.metadata.primitives.Contributor librettist:Marie Červinková-Riegrová>),
+        ('marcrel:CTB',
+        <music21.metadata.primitives.Contributor based on plot by:Ferdinand Mikovec>),
+        ('marcrel:CTB',
+        <music21.metadata.primitives.Contributor original incomplete plot by:Friedrich Schiller>)]
```

Comment:
Another thing to put a flag into for after this PR (I want to get this merged soon).  I'm thinking that things like 'marcrel:LBT' should be put into the Primitives object as a list of things that can be found in the object, and not emphasized in our code as much.  For instance, it's better to get a list of four Contributor objects and then have `Contributor.roleToCode()`  or `Contributor.roleToCode('marcrel')` or `Contributor.roleToCode('dc')` so that the objects themselves (based on role, etc.) can know what their tag is when outputting to various formats.  "based on plot by" would have `roleToCode('musicxml') == 'miscellaneous'` or `'<miscellaneous>'` or something like that.

For later.  I'm just not so 100% happy with having our format be tied to one export or another.

But yes, I am worried about my proposal being another example of: https://xkcd.com/927/